ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Send.hs

```diff
@@ -289,62 +298,69 @@ createNewReciever uiLangFace@UiLangFace{..} s@Send{..} = do
-  QWidget.adjustSize recieversWidget
+  modifyIORef receivers (r :)
+  QWidget.adjustSize receiversWidget
   QWidget.adjustSize send
   revalidate s
 
-removeReciever :: UiLangFace -> Reciever -> Send -> IO ()
-removeReciever uiLangFace@UiLangFace{..} Reciever{..} s@Send{..} = do
-  enabledRecievers <- getEnabledRecievers s
-  QWidget.setEnabled recieverWidget False
-  QWidget.setVisible recieverWidget False
+removeReceiver :: UiLangFace -> Receiver -> Send -> IO ()
+removeReceiver uiLangFace@UiLangFace{..} Receiver{..} s@Send{..} = do
+  enabledReceivers <- getEnabledReceivers s
+  QWidget.setEnabled receiverWidget False
+  QWidget.setVisible receiverWidget False
   revalidate s
   recalcTotal uiLangFace s
-  -- To prevent removing last reciever
-  if length enabledRecievers == 1
+  -- To prevent removing last receiver
+  if length enabledReceivers == 1
     then do
-      createNewReciever uiLangFace s
+      createNewReceiver uiLangFace s
     else do
-      QWidget.adjustSize recieversWidget
+      QWidget.adjustSize receiversWidget
       QWidget.adjustSize send
 
-getEnabledRecievers :: Send -> IO [Reciever]
-getEnabledRecievers Send{..} = do
-  r <- readIORef recievers
-  filterM (\Reciever{..} -> QWidget.isEnabled recieverWidget) r
+getEnabledReceivers :: Send -> IO [Receiver]
+getEnabledReceivers Send{..} = do
+  r <- readIORef receivers
+  filterM (\Receiver{..} -> QWidget.isEnabled receiverWidget) r
 
 recalcTotal :: UiLangFace -> Send -> IO ()
 recalcTotal UiLangFace{..} s@Send{..} = do
-  enabledRecievers <- getEnabledRecievers s
-  textAmounts <- mapM (\Reciever{..} -> fromString <$> QLineEdit.text amountEdit) enabledRecievers
-  let amounts = map Unsafe.fromJust (filter isJust (map (\a -> normalize <$> readMaybe a) textAmounts))
-  langPutUiCommand (UiCalcTotal amounts) >>= \case
-    Left err -> void $ QMessageBox.critical send ("Error" :: String) $ toString err
-    Right _ -> pass
-
-changeTotalAmount :: Send -> Text -> IO ()
-changeTotalAmount Send{..} amount = do
+  enabledReceivers <- getEnabledReceivers s
+  textAmounts <- mapM (fmap fromString . QLineEdit.text . amountEdit) enabledReceivers
+  let amounts = catMaybes (filter isJust (map (fmap normalize . readMaybe) textAmounts))
+  -- To prevent summarizing amounts, that typed improperly
+  if length enabledReceivers == length amounts
+    then do
+      QWidget.setStyleSheet totalAmount ("QLabel { color : #34343e }" :: String)
+      langPutUiCommand (UiCalcTotal amounts) >>= \case
+        Left err -> void $ QMessageBox.critical send ("Error" :: String) $ toString err
+        Right _ -> pass
+    -- Color amount gray, when something is typed improperly. Last correct total amount will be shown
+    else QWidget.setStyleSheet totalAmount ("QLabel { color : gray }" :: String)
+
+changeTotalAmount :: Send -> Text -> Text -> IO ()
+changeTotalAmount Send{..} amount unit = do
   QLabel.setText totalAmount $ toString amount
+  QLabel.setText totalUnit $ toString unit
```

Comment:
Btw, did you check that it looks OK when unit is "Lovelace" (which is longer than "ADA")?
Also, I've seen a screenshot where you have something-something `e12`. Please avoid scientific stuff in user-visible strings, if possible.

Reply:
![image](https://user-images.githubusercontent.com/14182218/48422696-41bc1a00-e770-11e8-8328-2efbed3be1b0.png)
looks good, I think